### DISCUSSION TITLE
eslint rule to prevent imports from packages to app

### DISCUSSION THREAD

**Discussion by richardshephard1** [vercel/turborepo#9230] [Help] [answered]
### Summary

We have a mono-repo with a NextJS webapp and a UI library within packages. In the UI library it is possible to import:
```js
// packages/ui-components/src/page-intro.tsx
import { Alpha } from '../../../apps/web/app/_services/models/page-intro';
```

This imports types from the NextJS app, which doesn't match our architecture. We only allow NextJS to import from the UI components, not the other way around. 

Is there any eslint/lint rule that can be enabled such that it can flag developers that they have imported files incorrectly?

**Comment by anthonyshew** [comment, marked as answer]
This sounds like this rule, if I'm understanding correctly:  https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-relative-packages.md
